src/remote-control/process-update-delegate.js

```diff
@@ -177,11 +262,75 @@ export default class ProcessUpdateDelegate {
             }
         });
 
+        const { inMeeting } = getInMeetingStatus(this._store.getState());
+
+        if (inMeeting && !newState.inMeeting) {
+            this.stopScreenshare();
+        }
+
         this._store.dispatch(updateSpotState(newState));
 
         return Promise.resolve();
     }
 
+    /**
+     * Begins the screensharing connection establishment processes between a
+     * remote control and a Jitsi-Meet meeting.
+     *
+     * @param {string} spotId - The jid of the Spot to use as a signaling layer
+     * for establishing a screenshare connection. Messages go to Spot for it
+     * to pass into the meeting.
+     * @param {string} roomFullJId - The jid of the remote control that wants
+     * to establish a {@code ScreenshareConnection}.
+     * @param {Object} screenshareConnection - The instance of
+     * {@code ScreenshareConnection} to be used.
+     * @returns {Promise}
+     */
+    startScreenshare(spotId, roomFullJId, screenshareConnection) {
+        const { screensharing } = getInMeetingStatus(this._store.getState());
+
+        if (screensharing) {
+            logger.error('Tried to start screenshare while already started.');
+
+            return Promise.reject();
+        }
+
+        this._screenshareConnection = screenshareConnection;
+
+        this._store.dispatch(updateSpotState({
+            isWirelessScreenshareConnectionActive: true
+        }));
+
+        return screenshareConnection.startScreenshare(
+            spotId,
+            roomFullJId
+        ).catch(error => {
```

Comment:
dumb question: if you catch here, does the failure still get propagated up to the promise that was returned? (i.e., if the caller has a catch?) or do we not want that?

Reply:
It won't get propagated up. At some point if I don't catch  itand handle it an error will be logged about an uncaught promise rejection. I can double check swallowing the error is in fact what I want.